posts/process-csvs-from-amazon-s3-using-apache-flink-jhipster-and-kubernetes.md

```diff
@@ -0,0 +1,187 @@
+---
+authors:
+- Theo Lebrun
+tags:
+- Big Data
+- Data Streaming
+- AWS
+- Kubernetes
+date: 2021-01-31T14:50:55.000Z
+title: "Process CSVs from Amazon S3 using Apache Flink, JHipster and Kubernetes"
+image: 
+---
+
+[Apache Flink](https://flink.apache.org/) is one of the latest distributed Big Data frameworks with a goal of replacing Hadoop's MapReduce. [Apache Spark](https://spark.apache.org/) is "very" similar to Flink but where Flink shines is by being able to process streams of data in real time. Spark on the other hand can only do batch processing and lack of stream processing capabilities. Real time data streaming is now basically everywhere and a lot of companies are upgrading their existing infrastructure to support it. If you are looking to stream data or simply want to integrate a new framework built for data streaming, Apache Flink is a real option.
+
+In this blog post, I will explain how a local Flink cluster running on Kubernetes can be used to process data stored on Amazon S3. The data is coming from [the CDC](https://data.cdc.gov/browse?tags=covid-19) website and the goal is to join them to correlate the number of vaccine doses with COVID-19 cases/deaths. All the code used in this blog post is available at this [GitHub repository](https://github.com/Falydoor/blog-flink).
+
+# Deployment on Kubernetes
+
+## Prerequisites
+
+I decided to use [minikube](https://minikube.sigs.k8s.io/docs/start/) for my local Kubernetes cluster but you can also use a Cloud service like [Amazon EKS](https://aws.amazon.com/eks/) or [GKE](https://cloud.google.com/kubernetes-engine). Make sure to have your `kubectl` all configured to interact with your cluster correctly. In case of any issues, you can follow the instructions from the [Flink's website](https://ci.apache.org/projects/flink/flink-docs-release-1.12/deployment/resource-providers/standalone/kubernetes.html#starting-a-kubernetes-cluster-session-mode).
```

Comment:
```suggestion
I decided to use [minikube](https://minikube.sigs.k8s.io/docs/start/) for my local Kubernetes cluster but you can also use a Cloud service like [Amazon EKS](https://aws.amazon.com/eks/) or [GKE](https://cloud.google.com/kubernetes-engine). Make sure to have your `kubectl` configured to interact with your cluster correctly. In case of any issues, you can follow the instructions from the [Flink website](https://ci.apache.org/projects/flink/flink-docs-release-1.12/deployment/resource-providers/standalone/kubernetes.html#starting-a-kubernetes-cluster-session-mode).
```